igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerRegistryCache.groovy

```diff
@@ -0,0 +1,92 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.docker
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import redis.clients.jedis.Jedis
+import redis.clients.jedis.JedisPool
+
+@SuppressWarnings(['PropertyName', 'DuplicateNumberLiteral'])
+@Service
+class DockerRegistryCache {
+    @Autowired
+    JedisPool jedisPool
+
+    String id = 'dockerRegistry'
+
+    @SuppressWarnings('GStringExpressionWithinString')
+    @Value('${spinnaker.jedis.prefix:igor}')
+    String prefix
+
+    List<String> getImages(String account) {
+        Jedis resource = jedisPool.resource
+        def key = makeKey(account, '*', '*', '*')
+
+        resource.keys(key).collect {
+            def parse = parseKey(it)
+            if (parse) {
+                formatImage(parse.registry, parse.repository, parse.tag)
+            } else {
+                null
+            }
+        }.unique()
+    }
+
+    String getLastDigest(String account, String registry, String repository, String tag) {
+        Jedis resource = jedisPool.resource
+        def key = makeKey(account, registry, repository, tag)
+        if (!resource.exists()) {
+            return [:]
+        }
+
+        Map result = resource.hgetAll(key)
+        result.digest
+    }
+
+    void setLastDigest(String account, String registry, String repository, String tag, String digest) {
+        Jedis resource = jedisPool.resource
+        String key = makeKey(account, registry, repository, tag)
+        resource.hset(key, 'digest', digest)
+        jedisPool.returnResource(resource)
+    }
+
+    void remove(String account, String registry, String repository, String tag) {
+        Jedis resource = jedisPool.resource
+        resource.del(makeKey(account, registry, repository, tag))
+        jedisPool.returnResource(resource)
+    }
+
+    private String makeKey(String account, String registry, String repository, String tag) {
+        "${prefix}:${id}:${account}:${registry}:${repository}:${tag}"
+    }
+
+    private Map<String, String> parseKey(String key) {
+        def split = key?.split("'")
+
+        if (!split || split[0] != prefix || split[1] != id || split.length != 6) {
+            return null
+        }
+
+        [account: split[2], registry: split[3], repository: split[4], tag: split[5]]
+    }
+
+    static public String formatImage(String registry, String repository, String tag) {
+        registry + '/' + repository + ':' + tag // for some reason groovy string magic doesn't work here.
```

Comment:
Not sure what you mean it 'doesn't work here'. If you tried to use the resulting value as a map key, and a subsequent lookup failed in that map failed, it's likely because the resulting value is not of type `String` (it will be a `GString`).
Try this: `"$registry/$repository:$tag".toString()"`